### PR TITLE
efibootmgr needs separate disk and partition

### PR DESCRIPTION
**Pull request by pajamian**
efibootmgr needs a separate device (/dev/xxx) and partition number.

### CODE REVIEW

**Comment by pajamian** [comment]
This should resolve #49

File: migrate2rocky/migrate2rocky.sh [hunk 1]
```diff
@@ -351,10 +351,13 @@ collect_system_info () {
...
     # other checks if it's not there.
     if [[ $update_efi ]]; then
-	declare -g efi_mount
+	local efi_mount
+	declare -g efi_disk efi_partition
 	efi_mount=$(findmnt --mountpoint /boot/efi --output SOURCE \
 	    --noheadings) ||
 	    exit_message "Can't find EFI mount.  No EFI  boot detected."
+	efi_disk=/dev/$(lsblk -no pkname "$efi_mount")
+	efi_partition=$(<"/sys/block/$efi_disk/${efi_mount##*/}/partition")
```
Comment (McNutnut) [inline migrate2rocky/migrate2rocky.sh:360]:
```
[root@server ~]# efi_partition=$(<"/sys/block/$efi_disk/${efi_mount##*/}/partition")
-bash: /sys/block//dev/sdb/sdb2/partition: No such file or directory
```

An additional `/dev` needs to be stripped,

```suggestion
	efi_partition=$(<"/sys/block/${efi_disk##*/}/${efi_mount##*/}/partition")
```

**Review by McNutnut — CHANGES_REQUESTED** [review]
Definitely a better approach!

File: migrate2rocky/migrate2rocky.sh (hunk 1 above)
Comment (pajamian) [reply migrate2rocky/migrate2rocky.sh]: Thanks, good spotting.

**Pull request merged**